service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/OutputMessageWrapperMapperTest.java

```diff
@@ -18,44 +20,71 @@
-    private CharSequence expectedOutputMessage;
+    private GetGpcStructuredTaskDefinition gpcStructuredTaskDefinition;
 
     @BeforeEach
-    public void setUp() throws IOException {
+    void setUp() throws IOException {
         when(randomIdGeneratorService.createNewId()).thenReturn(TEST_ID);
         when(timestampService.now()).thenReturn(Instant.parse(TEST_DATE_TIME));
         outputMessageWrapperMapper = new OutputMessageWrapperMapper(randomIdGeneratorService, timestampService);
-
-        getGpcStructuredTaskDefinition = GetGpcStructuredTaskDefinition.builder()
+        gpcStructuredTaskDefinition = GetGpcStructuredTaskDefinition.builder()
             .toAsid(TO_ASID_VALUE)
             .fromAsid(FROM_ASID_VALUE)
             .build();
+    }
+
+    @Test
+    void When_MappingOutputMessageWrapperWithStringContentAndRedactionsDisabled_Expect_UK06InteractionIdToBePresentAndProperXmlOutput() {
+        final String expected = ResourceTestFileUtils.getFileContent(EXPECTED_OUTPUT_MESSAGE_WRAPPER_NO_REDACTIONS_XML);
+        final String actual = outputMessageWrapperMapper.map(gpcStructuredTaskDefinition, TRANSFORMED_EXTRACT);
 
-        expectedOutputMessage = ResourceTestFileUtils.getFileContent(EXPECTED_OUTPUT_MESSAGE_WRAPPER_XML);
+        assertAll(
+            () -> assertThat(actual).isEqualToIgnoringWhitespace(expected),
+            () -> assertThat(actual).contains(EHR_EXTRACT_INTERACTION_ID_NO_REDACTIONS),
+            () -> assertThat(actual).doesNotContain(EHR_EXTRACT_INTERACTION_ID_WITH_REDACTIONS)
+        );
     }
 
     @Test
-    public void When_MappingOutputMessageWrapperWithStringContent_Expect_ProperXmlOutput() {
-        String outputMessage = outputMessageWrapperMapper.map(
-            getGpcStructuredTaskDefinition,
-            TRANSFORMED_EXTRACT);
+    void When_MappingOutputMessageWrapperWithStringContentAndRedactionsEnabled_Expect_UK07InteractionIdToBePresentAndProperXmlOutput()
+        throws IllegalAccessException, NoSuchFieldException {
+        final String expected = ResourceTestFileUtils.getFileContent(EXPECTED_OUTPUT_MESSAGE_WRAPPER_WITH_REDACTIONS_XML);
+
+        enableRedactions();
+
+        final String actual = outputMessageWrapperMapper.map(gpcStructuredTaskDefinition, TRANSFORMED_EXTRACT);
+
+        assertAll(
+            () -> assertThat(actual).isEqualToIgnoringWhitespace(expected),
+            () -> assertThat(actual).contains(EHR_EXTRACT_INTERACTION_ID_WITH_REDACTIONS),
+            () -> assertThat(actual).doesNotContain(EHR_EXTRACT_INTERACTION_ID_NO_REDACTIONS)
+        );
+    }
 
-        assertThat(outputMessage).isEqualToIgnoringWhitespace(expectedOutputMessage);
+    private void enableRedactions() throws IllegalAccessException, NoSuchFieldException {
+        final Field field = outputMessageWrapperMapper.getClass().getDeclaredField("redactionsEnabled");
+        field.setAccessible(true);
+        field.set(outputMessageWrapperMapper, true);
```

Comment:
I'm not a fan of code which requires reflection to be tested. I would describe that code as 'untestable', or at the very least 'not nice to test'. We want code which encourages developers to write tests, not puts them off.

Reply:
Ah yes, I forgot I mentioned yesterday I'll take an alternative approach - thanks 😄